### DISCUSSION TITLE
Building a Minimal ASP.NET Core clone

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#77] [General]
# Building a Minimal ASP.NET Core clone

In this article, we will build a minimal version of what ASP.NET Core does - yes, you read right. We will create a very simplistic clone of ASP.NET Core to discuss how the whole thing works. Beginning with a simple console application, we will add the necessary components to make it work as a web server. Bonus points for our own middleware pipeline and dependency injection.

https://steven-giesel.com/blogPost/e84dec43-51f1-4c70-8b3b-dcdb1c3164ce

**Comment by samcov** [comment]
This is very interesting, good job!

**Reply by linkdotnet** [reply]
Thanks - was also interesting to try out in the first place :D

**Comment by vinhlq289** [comment]
Thanks
